src/modules/checkout/templates/checkout-form/index.tsx

```diff
@@ -17,9 +17,20 @@ export default async function CheckoutForm({
     return null
   }
 
-  const shippingMethods = await listCartShippingMethods(cart.id)
+  const _shippingMethods = await listCartShippingMethods(cart.id)
   const paymentMethods = await listCartPaymentMethods(cart.region?.id ?? "")
 
+  const shippingMethods = _shippingMethods.filter((sm) => {
+    const isReturn = sm.rules.some(
+      (rule) => rule.attribute === "is_return" && rule.value === "true"
+    )
+    const isEnabledInStore = sm.rules.some(
+      (rule) => rule.attribute === "enabled_in_store" && rule.value === "true"
+    )
```

Comment:
thought: This should probably be a fixed filter on the backend when you are using `GET /store/shipping-options` – wdyt?

Reply:
I added the check here because of the comment about the return flow on linear, but I can move this to the core!

Reply:
Yes, will push a change 👍